instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/tracers/graphql_tracer.rb

```diff
@@ -73,6 +85,16 @@ def config
           def attributes_for(key, data)
             attributes = {}
             case key
+            when 'execute_field', 'execute_field_lazy'
+              attributes['graphql.field.parent'] = data[:owner].graphql_name # owner is the concrete type, not interface
+              attributes['graphql.field.name'] = data[:field].graphql_name
+              attributes['graphql.lazy'] = key == 'execute_field_lazy'
+            when 'authorized', 'authorized_lazy'
+              attributes['graphql.type.name'] = data.fetch(:type).graphql_name
```

Comment:
Is it possible for the `data` hash here to not contain the key `:type`? Fetch will raise if it is absent, which I believe in this code path would introduce a runtime exception.

Reply:
I believe it is fine as it would have raised in `PlatformTracing#trace` before now. [Ever since that `key` code path has been added for authorized and resolve_type](https://github.com/rmosolgo/graphql-ruby/pull/2660/files#diff-957133c12f5c1bb43a25cf51171630b8b7baff115db50158543d40f2b938e615R55-R72), it uses `fetch` so I feel safe with this.

Reply:
I find myself wondering why we're using `fetch` here at all, in fact. I would change it to either:
- Just grab it from the hash and use safe nav: `attributes['graphql.type.name'] = data[:type]&.graphql_name`
- Only add it if we know it's present: `attributes['graphql.type.name'] = data[:type].graphql_name if data[:type]`

Reply:
Unless `data` isn't a hash, of course.

Reply:
[Docs](https://graphql-ruby.org/api-doc/2.0.16/GraphQL/Tracing) covering `data` contents. I believe it is a hash.

Reply:
If it is programmatically impossible for some of these values to not be present we probably don't need to introduce defensive programming, but we do need to follow the error handling guidelines https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/error-handling.md#basic-error-handling-principles

My personal take here is that I'm inclined to err to the side of caution.